geoalchemy2/elements.py

```diff
@@ -114,10 +117,10 @@ class WKTElement(_SpatialElement):
 
     _REMOVE_SRID = re.compile("(SRID=([0-9]+); ?)?(.*)")
 
-    geom_from = "ST_GeomFromText"
-    geom_from_extended_version = "ST_GeomFromEWKT"
+    geom_from: str = "ST_GeomFromText"
+    geom_from_extended_version: str = "ST_GeomFromEWKT"
 
-    def __init__(self, data, srid=-1, extended=None):
+    def __init__(self, data, srid: int = -1, extended: Optional[bool] = None) -> None:
```

Comment:
Why not adding type annotation for `data`? While under certain conditions (i.e. when `extended=False` of when `extended=True and srid != -1`) it can actually work with anything that can implicitly cast to `str`, it is strongly advised to pass an actual `str` object as `data` argument.

Reply:
I wasn't sure what exactly to do here with the type of `data` since the following code seems to suggest that it must be a `str` type:
```python
class WKTElement(_SpatialElement):
    """Instances of this class wrap a WKT or EWKT value.

    Usage examples::

        wkt_element_1 = WKTElement('POINT(5 45)')
        wkt_element_2 = WKTElement('POINT(5 45)', srid=4326)
        wkt_element_3 = WKTElement('SRID=4326;POINT(5 45)', extended=True)
    """

    _REMOVE_SRID = re.compile("(SRID=([0-9]+); ?)?(.*)")

    geom_from: str = "ST_GeomFromText"
    geom_from_extended_version: str = "ST_GeomFromEWKT"

    def __init__(self, data, srid: int = -1, extended: Optional[bool] = None) -> None:
        if extended is None:
            extended = data.startswith("SRID=")
        if extended and srid == -1:
            # read srid from EWKT
            data_s = data.split(";")
            if len(data_s) != 2:
                raise ArgumentError("invalid EWKT string {}".format(data))
            header = data_s[0]
            try:
                srid = int(header[5:])
            except ValueError:
                raise ArgumentError("invalid EWKT string {}".format(data))
        _SpatialElement.__init__(self, data, srid, extended)
```

In this case whatever the type is for the `data` parameter it must support the `.startswith` and `.split` methods which seems to indicate that it needs to just be a `str`. We can add a direct annotation on `str` if that works.

Reply:
Yeah for `WKTElement` we should only use `str` objects.
For `WKBElement` I think we can accept `str` and `memoryview`.

Reply:
See commits 3be4ed5 and b40ca22

Reply:
Perfect, thanks!